### DISCUSSION TITLE
Date Popup

### DISCUSSION THREAD

**Discussion by shakamoushie** [zauberzeug/nicegui#263] [Q&A] [answered]
I couldn't find a date popup. Are there plans to add this widget (with an initial blank date)?

Thanks in advance..

**Comment by falkoschindler** [comment, marked as answer]
@shakamoushie You mean something like [QDate](https://quasar.dev/vue-components/date)?
We haven't thought about it. But it would certainly a useful feature.

Until it will be included in NiceGUI, you can build your own with a generic `Element`:

```python
from nicegui import ui
from nicegui.element import Element

def handle_change(msg: dict) -> None:
    print(msg['args'], flush=True)
    date.props(f'model-value={msg["args"]}')

date = Element('q-date').props('model-value=2023/01/01').on('update:model-value', handle_change)

ui.run()
```

**Comment by shakamoushie** [comment]
Fantastic @falkoschindler, This is so awesome. Thank you very much :-)
